nifi-nar-bundles/nifi-salesforce-bundle/nifi-salesforce-processors/src/main/java/org/apache/nifi/processors/salesforce/QuerySalesforceObject.java

```diff
@@ -0,0 +1,486 @@
+                AGE_FIELD,
+                INITIAL_AGE_FILTER,
+                AGE_DELAY,
+                API_VERSION,
+                BASE_URL,
+                AUTH_SERVICE,
+                RECORD_WRITER,
+                INCLUDE_ZERO_RECORD_FLOWFILES,
+                RESPONSE_TIMEOUT
+        ));
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        final Set<Relationship> relationships = new HashSet<>();
+        relationships.add(REL_SUCCESS);
+        return relationships;
+    }
+
+    @Override
+    protected Collection<ValidationResult> customValidate(ValidationContext validationContext) {
+        final List<ValidationResult> results = new ArrayList<>(super.customValidate(validationContext));
+        if (validationContext.getProperty(INITIAL_AGE_FILTER).isSet() && !validationContext.getProperty(AGE_FIELD).isSet()) {
+            results.add(
+                    new ValidationResult.Builder()
+                            .subject(INITIAL_AGE_FILTER.getDisplayName())
+                            .valid(false)
+                            .explanation("it requires " + AGE_FIELD.getDisplayName() + " also to be set.")
+                            .build()
+            );
+        }
+        return results;
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        String sObject = context.getProperty(SOBJECT_NAME).getValue();
+        String fields = context.getProperty(FIELD_NAMES).getValue();
+        String customWhereClause = context.getProperty(CUSTOM_WHERE_CONDITION).getValue();
+        RecordSetWriterFactory writerFactory = context.getProperty(RECORD_WRITER).asControllerService(RecordSetWriterFactory.class);
+        boolean includeZeroRecordFlowFiles = context.getProperty(INCLUDE_ZERO_RECORD_FLOWFILES).asBoolean();
+
+        String ageField = context.getProperty(AGE_FIELD).getValue();
+        String initialAgeFilter = context.getProperty(INITIAL_AGE_FILTER).getValue();
+        Long ageDelayMs = context.getProperty(AGE_DELAY).asTimePeriod(TimeUnit.MILLISECONDS);
+
+        String ageFilterLower;
+        StateMap state;
+        try {
+            state = context.getStateManager().getState(Scope.CLUSTER);
+            ageFilterLower = state.get(LAST_AGE_FILTER);
+        } catch (IOException e) {
+            throw new ProcessException("Last Age Filter state retrieval failed", e);
+        }
+
+        String ageFilterUpper;
+        if (ageField == null) {
+            ageFilterUpper = null;
+        } else {
+            Instant ageFilterUpperTime;
+            if (ageDelayMs == null) {
+                ageFilterUpperTime = Instant.now();
+            } else {
+                ageFilterUpperTime = Instant.now().minus(ageDelayMs, ChronoUnit.MILLIS);
+            }
+            ageFilterUpper = DateTimeFormatter.ISO_OFFSET_DATE_TIME
+                    .withZone(ZoneId.systemDefault())
```

Comment:
Is setting the Zone required? If the Offset Date Time format is desired, then `Instant.now()` should be replaced with `OffsetDateTime.now()` and then `OffsetDateTime.toString()` can be used without a DateTimeFormatter.

Reply:
Setting the zone is always required when DateTimeFormatter is used. The OffsetDateTime is a good idea, thanks!